### DISCUSSION TITLE
Sending files over the GenericHub

### DISCUSSION THREAD

**Discussion by lukeclements** [nasa/fprime#2308] [Q&A] [answered]
Hi, 

I have two deployments that use the GenericHub component and I was wondering how do I send flies over the GenericHub?

**Comment by LeStarch** [comment, marked as answer]
`Svc::FileDownlink` has a buffer send port.  This can be attached directly to the hub on one of its buffer input ports (don't forget to attach the return buffer port of the same index back to file downlink).

What happens on the host computer is up to the project:
1. You could attach the buffer output ports to `Svc::ComQueue` and the downlink stack.  That would push all file downlink buffers to the ground immediately.
2. You could add in `Svc::FileUplink` and wire that to the hub.  `Svc::FileDownlink` and `Svc::FileUplink` are designed to cross-wire, so you could "uplink" across the hub.  This would write the files locally and then another instance of `Svc::FileDownlink` could send to the com stack.

**Reply by lukeclements** [reply]
Thank you for the information!
